orchagent/fdborch.cpp

```diff
@@ -1201,12 +1379,38 @@ bool FdbOrch::addFdbEntry(const FdbEntry& entry, const string& port_name,
 
     FdbData storeFdbData = fdbData;
     storeFdbData.bridge_port_id = port.m_bridge_port_id;
+    // overwrite the type and origin
+    if ((fdbData.origin == FDB_ORIGIN_MCLAG_ADVERTIZED) && (fdbData.type == "dynamic_local"))
+    {
+        //If the MAC is dynamic_local change the origin accordingly
+        //MAC is added/updated as dynamic to allow aging.
+        storeFdbData.origin = FDB_ORIGIN_LEARN;
```

Comment:
@pettershao-ragilenetworks: Thanks for the comment. Once the MAC is converted to Local, Remote FDB will not be expected from ICCPD and function remoteFdbEntry will not get called. MAC is supposed to be aged out via the event SAI_FDB_EVENT_AGED. Hope this clarifies.

Reply:
but from my testing, removeFdbEntry will be called:
```
Jul 30 07:30:15.847477 sonic INFO swss#orchagent: :- removeFdbEntry: 1633: FdbOrch RemoveFDBEntry: mac=00:10:94:00:00:33 fdb origin is different; found_origin:1 delete_origin:8
```

Reply:
@pettershao-ragilenetworks: Since the MAC is updated as local MAC, the origin is set to LOCAL. When the MAC moves from remove to local, ICCPD further updates the MCLAG_FDB_TABLE to delete the MAC entry added earlier. In the case of  remove FDB called with different origin than existing origin is always ignored. This is TRUE for any MAC type L2, VxLAN, or MCLAG.

Reply:
so, when remote fdb added, and remote fdb deleted(remote peer flush fdb maybe), local peer will not remove the remote fdb entry by event but by age event, right?
```
remove
```

> so fdb entry syncd from remote peer will not be remoted immediately when remote peer do fdb flush action, right?
> @pettershao-ragilenetworks: Yes, If the ICCPD adds the MAC as local ( in certain scenarios like Iccpd session down..etc) the MAC entry is as good as Local HW learned. Remote peer flush will not cause delete , it can only be deleted by Local AGE or FLUSH events.

but the remote fdb is programed as STATIC, AGE or FLUSH will not remove them, right?